openlibrary/plugins/openlibrary/js/realtime_account_validation.js

```diff
@@ -1,12 +1,14 @@
 export function initRealTimeValidation() {
     const signupForm = document.querySelector('form[name=signup]');
     const i18nStrings = JSON.parse(signupForm.dataset.i18n);
+    const countInvalidInputs = () => document.querySelectorAll('form[name=signup] input.invalid').length;
 
     if (window.grecaptcha) {
         // Callback that is called when grecaptcha.execute() is successful
         // Checks whether reportValidity exists for cross-browser compatibility
+        // Includes invalid input count to account for checks not covered by reportValidity
         function submitCreateAccountForm() {
-            if (!signupForm.reportValidity || signupForm.reportValidity()) {
+            if ((!signupForm.reportValidity || signupForm.reportValidity()) && !countInvalidInputs()) {
```

Comment:
The boolean is getting a little complicated ; something like this might be easier! Assuming I did all the bool maths correctly :P
```suggestion
            const invalidInputs = document.querySelectorAll('form[name=signup] input.invalid').length;
            const isFormValid = !signupForm.reportValidity || signupForm.reportValidity();
            if (isFormValid && !invalidInputs) {
```